server/src/test/java/org/elasticsearch/search/aggregations/AggregationCollectorTests.java

```diff
@@ -19,54 +19,109 @@
 
 package org.elasticsearch.search.aggregations;
 
-import org.elasticsearch.common.xcontent.XContentParser;
-import org.elasticsearch.common.xcontent.json.JsonXContent;
-import org.elasticsearch.index.IndexService;
-import org.elasticsearch.search.aggregations.MultiBucketConsumerService.MultiBucketConsumer;
-import org.elasticsearch.search.aggregations.support.AggregationContext.ProductionAggregationContext;
-import org.elasticsearch.test.ESSingleNodeTestCase;
+import org.apache.lucene.index.DirectoryReader;
+import org.apache.lucene.index.LeafReaderContext;
+import org.apache.lucene.index.RandomIndexWriter;
+import org.apache.lucene.search.IndexSearcher;
+import org.apache.lucene.store.Directory;
+import org.elasticsearch.index.mapper.KeywordFieldMapper.KeywordFieldType;
+import org.elasticsearch.script.AggregationScript;
+import org.elasticsearch.script.Script;
+import org.elasticsearch.script.ScriptService;
+import org.elasticsearch.search.aggregations.bucket.terms.TermsAggregationBuilder;
+import org.elasticsearch.search.aggregations.metrics.TopHitsAggregationBuilder;
 
 import java.io.IOException;
 
+import static org.mockito.Matchers.any;
 import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
 
-public class AggregationCollectorTests extends ESSingleNodeTestCase {
+public class AggregationCollectorTests extends AggregatorTestCase {
+    public void testTermsDoesntNeedScores() throws IOException {
+        assertFalse(needsScores(termsBuilder().field("f")));
+    }
+
+    public void testSubTermsDoesntNeedScores() throws IOException {
+        assertFalse(needsScores(termsBuilder().field("f").subAggregation(new TermsAggregationBuilder("i").field("f"))));
+    }
+
+    public void testScriptDoesntNeedScoresIfScriptDoesntNeedScores() throws IOException {
+        assertFalse(needsScores(termsBuilder().script(new Script("no_scores"))));
+    }
+
+    public void testScriptNeedsScoresIfScriptNeedsScores() throws IOException {
+        assertTrue(needsScores(termsBuilder().script(new Script("with_scores"))));
+    }
 
-    public void testNeedsScores() throws Exception {
-        IndexService index = createIndex("idx");
-        client().prepareIndex("idx").setId("1").setSource("f", 5).execute().get();
-        client().admin().indices().prepareRefresh("idx").get();
+    public void testSubScriptDoesntNeedScoresIfSubScriptDoesntNeedScores() throws IOException {
```

Comment:
I am not a fan of this method of documenting tests. CamelCase sentences are hard to read, and with many similar names it's easy to miss the differences between two of them.  Maybe just name this something like `testSubScriptNoScore` and add a javadoc noting what it's intended to test?  Same for the other very long test names.

Reply:
I like having a test for each thing. I could go either way on the long names. I'll change them.